pkg/webservice/webservice.go

```diff
@@ -81,7 +82,7 @@ type YResponseWriter struct {
 
 func (rw *YResponseWriter) Write(bytes []byte) (int, error) {
 	rw.body = bytes
-	if rw.statusCode == -1 {
+	if rw.statusCode == -1 || (rw.statusCode >= http.StatusOK && rw.statusCode <= http.StatusIMUsed) {
```

Comment:
I still have concerns about checking the status code. This doesn't seem to be a reliable solution. I suggest reverting the wrapper change and continue to use the default writer, which is safer for me. If you look at https://golang.org/pkg/net/http/#Client.Get, the go client library will return an error with some internal logic and check over the HTTP status code. If the wrapper on the server-side wrongly operates on responses, we are in trouble.